### ISSUE TITLE
Failed to mine block, runtime error

### ISSUE THREAD

**Issue by baltar** [aeternity/epoch#1642]
I'm running v0.24.0 in the docker container, but mining doesn't seem to work.

## Expected Behavior

Mining should work.

## Actual Behavior

Mining doesn't seem to work, the epoch_mining.log repeatedly shows

> 2018-10-11 19:32:51.928 [error] <0.1243.0>@aec_conductor:handle_mining_reply:710 Failed to mine block, runtime error; retrying with different nonce (was 13728450104661644388). Error: {execution_failed,{signal,sigkill,false}}

## Steps to Reproduce the Problem

Follow the installation descriptions on https://github.com/aeternity/epoch/blob/v0.24.0/docs/docker.md
  
  
## Logs, error output, etc.
 
(see "Actual Behavior)

## Specifications

  - Virtualization: container 
  - Hardware specs: 
    * RAM: 16G (on the host, 2G in the container)
    * CPU: Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz
  - OS: MacOS 10.13.6 (host)
  - Node Version: v0.24.0

**Comment by dincho** [comment]
Hello @baltar,
could you please try to allocate at least 4G to the container?

Otherwise you should change to less memory intensive executable `lean30` see the [troubleshooting section](https://github.com/aeternity/epoch/wiki/Troubleshooting#node-failing-mining-attempts) for more details.

**Comment by kryztoval** [comment]
@dincho I got the same error, I set lean but now I get this on the epoch_mining log

`2018-10-25 10:52:08.940 [info] <0.4860.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:08.988 [info] <0.4928.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:09.111 [info] <0.4961.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:10.113 [info] <0.4997.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:10.254 [info] <0.5030.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:11.079 [info] <0.5069.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:11.106 [info] <0.1243.0>@aec_conductor:start_mining:666 Starting mining
2018-10-25 10:52:11.517 [info] <0.5103.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:11.739 [info] <0.5139.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:12.688 [info] <0.5208.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:12.808 [info] <0.5175.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:13.754 [info] <0.5245.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:13.928 [info] <0.5279.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:14.748 [info] <0.5316.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:15.138 [info] <0.5350.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:15.536 [info] <0.5384.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:16.349 [info] <0.5420.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:17.190 [info] <0.1910.0>@aec_conductor:post_block:138 Header failed validation: incorrect_pow
2018-10-25 10:52:17.191 [info] <0.19618.0>@aec_conductor:post_block:138 Header failed validation: incorrect_pow
2018-10-25 10:52:17.220 [info] <0.1958.0>@aec_conductor:post_block:138 Header failed validation: incorrect_pow
2018-10-25 10:52:17.256 [info] <0.1525.0>@aec_conductor:post_block:138 Header failed validation: incorrect_pow
2018-10-25 10:52:17.258 [info] <0.1640.0>@aec_conductor:post_block:138 Header failed validation: incorrect_pow
2018-10-25 10:52:17.270 [info] <0.1278.0>@aec_conductor:post_block:138 Header failed validation: incorrect_pow
2018-10-25 10:52:18.919 [info] <0.1863.0>@aec_conductor:post_block:138 Header failed validation: incorrect_pow
2018-10-25 10:52:19.377 [info] <0.5457.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:23.731 [info] <0.5493.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:25.326 [info] <0.5532.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:27.730 [info] <0.5572.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:29.614 [info] <0.5608.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:31.183 [info] <0.5644.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:32.637 [info] <0.5720.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:32.750 [info] <0.5754.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:32.873 [info] <0.5788.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:33.036 [info] <0.5822.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:33.082 [info] <0.5856.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:33.091 [info] <0.5890.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
2018-10-25 10:52:33.187 [info] <0.5924.1>@aec_conductor:add_synced_generation_batched:156 Header failed validation: incorrect_pow
`

**Comment by dincho** [comment]
@kryztoval This error means you're connected to incompatible (with your configuration) version.
You're either:
- not connected to the testnet
- your configuration does not match the testnet configuration.

I just tested following configuration:
```
$ cat ~/.aeternity/myepoch.yaml
---
keys:
    password: "top secret"
    dir: ./keys

chain:
    persist: true

mining:
    autostart: true
    beneficiary: "ak_2HNb8bivUoJTcaxD6VKDy2wPHvTuQgQRJ3dkF9RSyVFqBkMneC"
    cuckoo:
        miner:
            executable: lean30
            extra_args: ""
            node_bits: 30
```

I tested with docker image:
```
docker pull aeternity/epoch
docker run -it -p 3013:3013 \
  -v ~/.aeternity/myepoch.yaml:/home/epoch/myepoch.yaml \
  -e EPOCH_CONFIG=/home/epoch/myepoch.yaml \
  aeternity/epoch
```

**Comment by kryztoval** [comment]
My config had lean28 as stated in the how to mine configuration documentation under the mining section, but everything else was the same.

My configuration was
```
sync:
port: 3015
external_port: 3015
listen_address: 0.0.0.0

keys:
dir: keys
password: "XXXXXXXXXXXXXXXXXXXXXXXXX"

http:
external:
port: 3013
listen_address: 0.0.0.0
internal:
port: 3113
listen_address: 127.0.0.1

websocket:
channel:
port: 3014
listen_address: 0.0.0.0

mining:
beneficiary: "ak_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
autostart: true

chain:
persist: true
db_path: /home/epoch/node/data/
```


Now I simplified it to this 
```
keys:
dir: keys
password: "XXXXXXXXXXXXXXXXXXXXXXXXX"

mining:
beneficiary: "ak_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
autostart: true

chain:
persist: true
db_path: /home/epoch/node/data/
```
and I don't see the error anymore. could it be from the sync part?

**Comment by kryztoval** [comment]
Problem was the mining part, it was starting meanXX, and I switched to lean28 but it turns out the correct one for version 0.24 is lean30 instead.

After that pow errors disappeared.

so if you followed this https://github.com/aeternity/epoch/blob/master/docs/release-notes/RELEASE-NOTES-0.8.0.md
you need to set it to
```
mining:
    autostart: true
    cuckoo:
        miner:
            executable: lean30
            extra_args: ""
            node_bits: 30
```

**Comment by dincho** [comment]
@kryztoval Not sure why you follow the release notes of 0.8.0 while the latest version is 0.25.0. However, I'm glad your issue is resolved.

**Comment by kryztoval** [comment]
Because it was the top search result I got when searching for the answer in google. :(
I see there is an English FAQ that is at blog.aeternity.com and maybe also at aeternity.com but the issue here is that Spanish users can't get to that FAQ normally from aeternity,com

For instance, if I go to the main site it is impossible for me to see the English site, I can pick Spanish, Russian, Chinese. But if I click on English I am redirected to the Spanish website and the Spanish FAQ has nothing to do with the incredible English FAQ that contains all the links I needed to solve those questions.

When you click on blog in the Spanish aternity page you are taken to a Spanish tagged posts in blog.aeternity.com (media site).

**In short**, the page that needs updating is the Spanish FAQ for the aeternity.com/es website
And please make it so if you chose English from the dropdown it stays in English instead of autoswitching?

**Comment by j28** [comment]
@kryztoval thanks for pointing this out. We can adjust some settings in the language detection/application. It would be great if you could retest on our staging site after the changes. Where can we reach you to continue this thread in private?

**Comment by baltar** [comment]
> could you please try to allocate at least 4G to the container?
> 
> Otherwise you should change to less memory intensive executable `lean30` see the [troubleshooting section](https://github.com/aeternity/epoch/wiki/Troubleshooting#node-failing-mining-attempts) for more details.

I wanted to test this, but wanted to re-start with the newest version. Therefore I deleted the existing containers and recreated them according to https://github.com/aeternity/epoch/blob/v0.25.0/docs/docker.md

But epoch always immediately stops with this:

```
{validation_failed,[{error,[{data_invalid,#{<<"additionalProperties">> => false,<<"properties">> => #{<<"dir">> => #{<<"description">> => <<"Location (directory) of the public/private key pair(s)">>,<<"type">> => <<"string">>},<<"peer_password">> => #{<<"description">> => <<"Password used to encrypt the peer key-pair files - if left blank `password` will be used">>,<<"type">> => <<"string">>}},<<"type">> => <<"object">>},no_extra_properties_allowed,#{<<"dir">> => <<"./keys">>,<<"password">> => <<"password">>},[<<"keys">>,<<"password">>]}]}]}
```

Note that I did use just "password"  when generating the keys. 

I tried copying the keys from /tmp/generated to ~/.aeternity/keys but that didn't change anything. I also tried setting the keys directory to `/home/epoch/node/generated_keys` and adding `-v /tmp/generated_keys:/home/epoch/node/generated_keys` when running the container but no change. 

How is this supposed to work?

**Comment by kryztoval** [comment]
Please bear in mind that version 0.25 introduced a breaking configuration change.

The password field is now called "peer_password" while in version 0.24 it was just called "password"

It seems you are using "password=password" when you should either remove it or use "peer_password=password" instead.

**Comment by dincho** [comment]
@baltar It's configuration issue, could you please post your config ? You should replace  `keys.password` with `keys.peer_password`. The password property has been removed because of [throw away signing keys](https://github.com/aeternity/epoch/pull/1681).

**Comment by dincho** [comment]
Commenting race with @kryztoval - you won :)

**Comment by baltar** [comment]
Thanks @kryztoval and @dincho, that worked. I tested both using the lean30 miner and increasing ram with the default miner successfully. But since the default miner does not work with the default docker configuration of 2GB / container, I'd suggest changing the example configuration and adding a note on this to `docker.md`. I created a pull request for this here: https://github.com/aeternity/epoch/pull/1752
